packages/dds/sequence/src/sequence.ts

```diff
@@ -166,7 +166,7 @@ export abstract class SharedSegmentSequence<T extends ISegment>
     private readonly loadedDeferredIncomingOps: ISequencedDocumentMessage[] = [];
 
     private messagesSinceMSNChange: ISequencedDocumentMessage[] = [];
-    private readonly intervalMapKernel: MapKernel;
+    private readonly intervalCollections: DefaultMap<IntervalCollection<SequenceInterval>>;
```

Comment:
this is a good point to illustrate where i think we can reduce abstraction. The end state for me is that this is just IntervalCollection, no default map, and no generic.

Reply:
i could maybe be convinced that there is value in having layers here, if it enables unit testing or something, but i don't clearly see a case for it at the moment

Reply:
if we make that change it would accompany a change to sequence's public API. I'm guessing you're imagining if people previously used multiple collection keys (say "collection1", "collection2"), they might instead add those to the interval properties?